components/BottomSheet/examples/BottomSheetAutolayoutSafeAreaExample.m

```diff
@@ -0,0 +1,147 @@
+                               attribute:NSLayoutAttributeTop
+                              multiplier:1.0f
+                                constant:0.0f].active = YES;
+  [NSLayoutConstraint constraintWithItem:self.withinSafeAreaView
+                               attribute:NSLayoutAttributeLeading
+                               relatedBy:NSLayoutRelationEqual
+                                  toItem:self.view
+                               attribute:NSLayoutAttributeLeading
+                              multiplier:1.0f
+                                constant:0.0f].active = YES;
+  [NSLayoutConstraint constraintWithItem:self.withinSafeAreaView
+                               attribute:NSLayoutAttributeTrailing
+                               relatedBy:NSLayoutRelationEqual
+                                  toItem:self.view
+                               attribute:NSLayoutAttributeTrailing
+                              multiplier:1.0f
+                                constant:0.0f].active = YES;
+  // Don't expand into the safe area.
+  [NSLayoutConstraint constraintWithItem:self.withinSafeAreaView
+                               attribute:NSLayoutAttributeBottom
+                               relatedBy:NSLayoutRelationLessThanOrEqual
+                                  toItem:self.view
+                               attribute:NSLayoutAttributeBottom
+                              multiplier:1.0f
+                                constant:0.0f].active = YES;
+
+  // Set size of the withinSafeArea view.
+  [NSLayoutConstraint constraintWithItem:self.withinSafeAreaView
+                               attribute:NSLayoutAttributeWidth
+                               relatedBy:NSLayoutRelationEqual
+                                  toItem:nil
+                               attribute:NSLayoutAttributeNotAnAttribute
+                              multiplier:0.0f
+                                constant:SafeContentHeight].active = YES;
+  [NSLayoutConstraint constraintWithItem:self.withinSafeAreaView
+                               attribute:NSLayoutAttributeHeight
+                               relatedBy:NSLayoutRelationEqual
+                                  toItem:nil
+                               attribute:NSLayoutAttributeNotAnAttribute
+                              multiplier:0.0f
+                                constant:SafeContentHeight].active = YES;
+
+  self.preferredContentSize = CGSizeMake(SafeContentWidth, SafeContentHeight);
+}
+
+- (void)viewDidLayoutSubviews {
+  [super viewDidLayoutSubviews];
+  CGPoint origin = self.withinSafeAreaView.frame.origin;
+  UIWindow *keyWindow = [UIApplication mdc_safeSharedApplication].keyWindow;
+  CGPoint safeAreaViewAbsoluteOrigin = [keyWindow convertPoint:origin
+                                                      fromView:self.view];
+  CGFloat safeAreaBottomAbsoluteY =
+      safeAreaViewAbsoluteOrigin.y + self.withinSafeAreaView.bounds.size.height;
```

Comment:
You shouldn't access data directly from a CGRect that hasn't been run through CGRectStandardize().
Prefer CGRectGetHeight(rect) over rect.size.height for non-standardized rects.